src/CurlHandler.php

```diff
@@ -11,6 +11,30 @@
  * A handler that uses cURL to send the request.
  */
 class CurlHandler implements HttpHandlerInterface {
+    /**
+     * @var curl handle to be (re)used
+     */
+    protected $curl = null;
+
+    /**
+     * Gets a cURL resource for this request
+     *
+     * @param HttpRequest $request
+     * @return resource cURL handle
+     */
+    protected function getCurlHandle(HttpRequest $request) {
+        if ($this->curl === null) {
+            $this->curl = curl_init();
+        } elseif (!$request->hasHeader('connection') || strcasecmp($request->getHeader('connection'), 'close') === 0) {
+            return;
```

Comment:
This looks off.

Reply:
Indeed, that was debugging while I was verifying it was working. Good catch!